src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -588,9 +590,12 @@ trait Implicits {
       // We can only rule out a subtype relationship if the left hand
       // side is a class, else we may not know enough.
       case tr1 @ TypeRef(_, sym1, _) if sym1.isClass =>
+        def hasMember(tp: Type, name: Name) = {
```

Comment:
Could we narrow this helper method's scope a bit? `typeRefHasNoMember`? I fear a future refactoring could pull this out and miss that it's only valid under certain assumptions.